### DISCUSSION TITLE
How to bubble up test failures to CI when using the API?

### DISCUSSION THREAD

**Discussion by timbrinded** [vitest-dev/vitest#3145] [Q&A] [answered]
There is additional pre and post logic, so in my repo I have wrapped vitest, and am using it programatically (which works mostly quite well). 

However in my CI I need the ability to detect test failures and fail the job when there are test errors. 

Is there an idiomatic way of doing this? 

Looking at the core class I cant seem to find a suitable getter for the results:
https://github.com/vitest-dev/vitest/blob/0f86ff98975a80191d6215b0d30ff6cd9f8388d3/packages/vitest/src/node/core.ts#L28

**Comment by timbrinded** [comment]
I am currently doing this, but i thought there would be a better way:

```
  const result = await executeTests(env, additionalArgs);
  const buffer = fs.readFileSync(result.cache.results.getCachePath(), "utf-8");
  const { results } = JSON.parse(buffer);
  const failed = results.map((result) => result[1].failed).filter((status) => status === true);

  if (failed.length > 0) {
    process.exit(1);
  } else {
    process.exit(0);
  }
```

**Comment by sheremet-va** [comment, marked as answer]
[`vitest.state.getFiles()`](https://github.com/vitest-dev/vitest/blob/0f86ff98975a80191d6215b0d30ff6cd9f8388d3/packages/vitest/src/node/state.ts#L56) returns the state of all test files.

Results can be processed with [`getTests()`](https://github.com/vitest-dev/vitest/blob/0f86ff98975a80191d6215b0d30ff6cd9f8388d3/packages/runner/src/utils/tasks.ts#L8) to get the flat structure of all test functions.
